app/src/converter.js

```diff
@@ -2,17 +2,52 @@ import open from 'open';
 
 const os = require("os");
 const path = require('path');
+const fs = require("node:fs");
 
 const logger = require('./logger');
 const rpc = require('./weh-rpc');
 
 const exec_dir = path.dirname(process.execPath);
-let ffmpeg = path.join(exec_dir, "ffmpeg");
-let ffprobe = path.join(exec_dir, "ffprobe");
 
-if (os.platform() == "win32") {
-  ffmpeg += ".exe";
-  ffprobe += ".exe";
+const use_prebuilt_ffmpeg = fileExistsSync(ensureProgramExt(path.join(exec_dir, "ffmpeg")));
+const ffmpeg = findExecutableFullPath("ffmpeg", exec_dir);
+const ffprobe = findExecutableFullPath("ffprobe", exec_dir);
+
+if (!fileExistsSync(ffmpeg)) {
+  logger.error("ffmpeg not found. Please ensure if ffmpeg is installed and try again.");
+  process.exit(1);
+}
+if (!fileExistsSync(ffprobe)) {
+  logger.error("ffprobe not found. Please ensure if ffprobe is installed and try again.");
+  process.exit(1);
+}
+
+function findExecutableFullPath(programName, extraPath = "") {
+  programName = ensureProgramExt(programName);
+  const envPath = (process.env.PATH || '');
+  const pathArr = envPath.split(path.delimiter);
+  if (extraPath) {
+    pathArr.unshift(extraPath);
+  }
+  const foundExecutablePath = pathArr
+    .map((x) => path.join(x, programName))
+    .find((x) => fileExistsSync(x));
+  return foundExecutablePath || '';
```

Comment:
Return null or undefined, not empty string.

Reply:
Modified.

But BTW, this is for the consistency of return type actually, on purpose. I believe this is a better practice (learnt from TypeScript projects) in JavaScript to avoid some potential TypeError.

Reply:
> OK, modified.
> 
> But BTW, this is for the consistency of return type actually, on purpose. I believe this is a better practice (learnt from TypeScript projects) in JavaScript to avoid some potential TypeError.

I understand the logic, but Typescript has `string | undefined`, and in the specific case of js, js are passed by reference :) just nitpicking.

Thank you for the PR, I'll look at it asap.